flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/tieredstore/upstream/TieredStoreMode.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.tieredstore.upstream;
+
+/** The tiered store mode for {@link TieredStoreResultPartition}. */
+public interface TieredStoreMode {
+
+    /**
+     * The shuffle records will be accumulated to the finished buffer before writing to the
+     * corresponding tier and the accumulator is the tier of {@link TierType#IN_CACHE}.
+     */
+    enum TierType implements TieredStoreMode {
+        IN_CACHE,
+        IN_MEM,
+        IN_DISK,
+        IN_REMOTE,
+    }
+
+    /**
+     * Currently, only these tier combinations are supported. If the configured tiers is contained
+     * in the following combinations, an exception will be thrown.
+     */
+    enum SupportedTierCombinations implements TieredStoreMode {
+        MEMORY,
+        MEMORY_DISK,
+        MEMORY_DISK_REMOTE,
```

Comment:
JavaDoc is required for each enum value.

Reply:
Fixed.